docs/supported-tools.md

```diff
@@ -0,0 +1,271 @@
+in the configuration closure, or you can do so by adding a suppressions file:
+
+`pmd-rules.xml`
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  name="Novoda PMD rules"
+  xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+  xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+
+  ...
+
+  <!--region File exclusions-->
+  <exclude-pattern>.*\.kt</exclude-pattern>
+  <!--endregion-->
+
+  ...
+
+</ruleset>
+```
+
+## Findbugs
+[Findbugs](http://findbugs.sourceforge.net/) is a static analysis tool that looks for potential bugs in Java code. It does not support Kotlin.
+It can be used in both pure Java, and Android Java projects. It then only makes sense to have Findbugs enabled if you have Java code in your project.
+The plugin only runs Findbugs on projects that contain the Java or the Android plugin.
+
+### Configure Findbugs
+Enabling and configuring Findbugs for a project is done through the `findbugs` closure:
+
+```gradle
+findbugs {
+    toolVersion // A string, most likely '3.0.1' — the latest Findbugs release (for a long time)
+    exclude // A fileTree, such as project.fileTree('src/test/java') to exclude Java unit tests
+    excludeFilter // A file containing the Findbugs exclusions, e.g., teamPropsFile('static-analysis/findbugs-excludes.xml')
+    includeVariants { variant -> ... } // A closure to determine which variants (for Android) to include
+}
+```
+
+You can have multiple `exclude` statements.
+
+For more informations about Findbugs rules, refer to the [official website](http://findbugs.sourceforge.net/bugDescriptions.html).
+
+### Findbugs in mixed-language projects
+If your project mixes Java and Kotlin code, you most likely want to have an exclusion in place for all `*.kt` files. You can use the `exclude`
+in the configuration closure, or you can do so by adding a suppression to the filter file:
+
+`findbugs-excludes.xml`
+```xml
+<FindBugsFilter>
+
+  ...
+
+  <Match>
+    <Source name="~.*\.kt" />
+  </Match>
+
+  ...
+
+</FindBugsFilter>
+```
+
+## Detekt
+[Detekt](https://github.com/arturbosch/detekt) is a static analysis tool that looks for potential bugs and style violations in Kotlin code. It
+does not support Java. It can be used in both pure Kotlin, and Android Kotlin projects. It then only makes sense to have Detekt enabled if you
+have Kotlin code in your project. The plugin only runs Detekt on projects that contain the Kotlin or the Kotlin-Android plugin.
+
+### IMPORTANT: setup Detekt
+Unlike the other tools, the plugin **won't automatically add Detekt** to your project. If you forget to do it, the plugin will fail the build
+with an error like:
+
+```
+The detekt plugin is configured but not applied. Please apply the plugin in your build script.
```

Comment:
Do we really need to go into this much details? Above paragraph is I think really good but the duplication of the error message is redundant. I don't think this will be helpful to anybody.

Reply:
Reasoning: you get the error message and search for it. You get a hit, with the resolution steps. Everyone's happy.

Reply:
Then I would put a link to our docs in the error message. Rather than having it duplicated here. 
They would see the error and click the link, that's all. Don't have to search for something. 

Duplication is bad because the error may change and it is hard to maintain this copy to be the same.

Reply:
That's a good idea, tomorrow I will change #53 for that to change the text here _and_ add a link to this bit.

Reply:
I will postpone this to _after_ I have split out the tools docs, one per file, so I don't have to go back and change things again. I need #57 merged, and then this one merged, to be able to proceed with this.

Reply:
That sounds good.